framework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -0,0 +1,218 @@
+from utils import utils, randomUtils
+from .. import ValidationBase
+#Internal Modules End--------------------------------------------------------------------------------
+
+class PhysicsGuidedCoverageMapping(ValidationBase):
+  """
+    PhysicsGuidedCoverageMapping is a base class for validation problems
+    It represents the base class for most validation problems
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, specs, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    specs = super(PhysicsGuidedCoverageMapping, cls).getInputSpecification()
+    #specs.addSub(metricInput)
+    return specs
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'POSTPROCESSOR PhysicsGuidedCoverageMapping'
+    self.dynamicType = ['static','dynamic'] #  for now only static is available
+    self.acceptableMetrics = ["CDFAreaDifference", "PDFCommonArea", "STDReduction"] #  acceptable metrics
+    self.name = 'PhysicsGuidedCoverageMapping'
+    # self.pivotParameter = None
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parsed paramInput for this class.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    super()._handleInput(paramInput)
+
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it loads the
+      results to specified dataObject
+      @ In, inputIn, list, dictionary of data to process
+      @ Out, outputDict, dict, dictionary containing the post-processed results
+    """
+    dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
+    pivotParameter = self.pivotParameter
+    names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+    if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+      if 'dynamic' not in self.dynamicType: #self.model.dataType:
+        self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+
+    if pivotParameter:
+      if len(inputIn['Data'][0][-1]['time']) != len(list(evaluation.values())[0]):
+        self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+      if pivotParameter not in evaluation:
+        evaluation[pivotParameter] = inputIn['Data'][0][-1]['time']
```

Comment:
I'm confused here, Can the class handle dynamic situation?

Reply:
I don't know... pivotParameter is not used in my case, I don't actually understand how to use this pivotParameter.